examples/smartContracts/index.ts

```diff
@@ -8,17 +8,106 @@ import {
 import { IEvent } from '../../src/interfaces/IEvent';
 import { IReadData } from '../../src/interfaces/IReadData';
 import { WalletClient } from '../../src/web3/WalletClient';
-import { deploySmartContracts } from './deployer';
+import { awaitTxConfirmation, deploySmartContracts } from './deployer';
 import { Args } from '../../src/utils/arguments';
 import { readFileSync } from 'fs';
 import { MassaCoin } from '../../src/web3/MassaCoin';
+import { Client } from '../../src/web3/Client';
+import {
+  EventPoller,
+  ON_MASSA_EVENT_DATA,
+  ON_MASSA_EVENT_ERROR,
+} from '../../src/web3/EventPoller';
+import { INodeStatus } from '../../src/interfaces/INodeStatus';
 const path = require('path');
 const chalk = require('chalk');
 const ora = require('ora');
 
+const MASSA_EXEC_ERROR = 'massa_execution_error';
+
 const DEPLOYER_SECRET_KEY =
   'S1NA786im4CFL5cHSmsGkGZFEPxqvgaRP8HXyThQSsVnWj4tR7d';
 
+export async function withTimeoutRejection<T>(
```

Comment:
it looks like duplicated code with the sc-deployer. maybe you should refactor code to avoid duplicating functions

Reply:
I will export that from web3 under utils and make it available for the sc-depliyor to import it. Sounds ok ?

Reply:
yes